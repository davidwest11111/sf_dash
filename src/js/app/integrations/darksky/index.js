import axios from 'axios';
import jsonpAdapter from 'axios-jsonp';


export const URL_BASE = 'https://api.darksky.net/forecast';

export default class DarkSky {
  static getUrl = ({
    // all params required
    base = URL_BASE,
    key,
    lat,
    long,
    exclude = 'minutely,flags,currently',
    lang = 'en',
    units = 'auto',
  }) => `${base}/${key}/${lat},${long}?exclude=${exclude}&lang=${lang}&units=${units}`;

  static async fetchWeatherData({ lat, long, key, units }) {
    // NOTE: this is a jsonp call. there are no status codes or errors to parse like with XHR
    // try/catch errors in components that call this, so just return the response as-is
    const response = await axios.get(this.getUrl({ lat, long, key, units }), {
      adapter: jsonpAdapter, // gets around CORS
    });
    return response;
  }

  static async fetchAllLocationsWeatherData(locations, key, units) {
    const darkSkyUnits = this.convertUnitsForDarkSky(units);
    const responses = await axios.all(
      locations.map(({ lat, long }) => (
        this.fetchWeatherData({ lat, long, key, units: darkSkyUnits })
      )),
    );
    return responses;
  }

  static isValidKey = async key => {
    const response = await this.fetchWeatherData({ lat: 38, long: -122, key });
    return !!response?.data;
  }

  static convertUnitsForDarkSky = (units = 'auto') => {
    switch (units) {
      case ('f'):
        return 'us';
      case ('c'):
        return 'si';
      case ('auto'):
        return 'auto';
      default:
        throw new Error(`Invalid Unit type: ${units}`);
    }
  };
}
