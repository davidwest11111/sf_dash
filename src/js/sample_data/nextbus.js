// The data returned is actually in XML, this is just what the X2JS library outputs

export const SAMPLE_PREDICTION = {
  direction: {
    prediction: [
      {
        _epochTime: '1588198229803',
        _seconds: '296',
        _minutes: '4',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6609',
        _block: '1620',
        _tripTag: '9359239',
        $$hashKey: 'object:19',
      },
      {
        _epochTime: '1588198681179',
        _seconds: '747',
        _minutes: '12',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6599',
        _block: '1622',
        _tripTag: '9359240',
        $$hashKey: 'object:20',
      }, { _epochTime: '1588199412858', _seconds: '1479', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6547', _block: '1633', _tripTag: '9359237', $$hashKey: 'object:21' }, { _epochTime: '1588199419316', _seconds: '1486', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6533', _block: '1628', _tripTag: '9359243', $$hashKey: 'object:22' }],
    _title: 'Inbound to Ferry Plaza',
  },
  message: [
    {
      _text: 'Service frequency for this stop is 10 minutes or less.',
      _priority: 'Normal',
    },
    {
      _text: 'Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info.',
      _priority: 'Low',
    }, { _text: 'Text COVID19SF to 888-777 for official updates. Visit sfdph.org', _priority: 'Low' }, { _text: 'Board thru rear doors. Front door for accessibility needs only.', _priority: 'Normal' }],
  _agencyTitle: 'SF Muni',
  _routeTitle: '14-Mission',
  _routeTag: '14',
  _stopTitle: 'Mission St & 24th St',
  _stopTag: '5565',
  $$hashKey: 'object:17',
  className: '',
};

// FIXME
export const SAMPLE_FAILED_PREDICTIONS = {

};

export const SAMPLE_PREDICTIONS_XML = `
<?xml version="1.0" encoding="utf-8" ?> 
<body copyright="All data copyright San Francisco Muni 2020.">
  <predictions agencyTitle="SF Muni" routeTitle="12-Folsom-Pacific" routeTag="12" stopTitle="Pacific Ave &amp; Grant Ave" stopTag="5837" dirTitleBecauseNoPredictions="Inbound to Van Ness">
    <message text="Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info." priority="Low"/>
    <message text="Text COVID19SF to 888-777 for official updates. Visit sfdph.org" priority="Low"/>
  </predictions>
  <predictions agencyTitle="SF Muni" routeTitle="14-Mission" routeTag="14" stopTitle="Mission St &amp; 24th St" stopTag="5565">
    <direction title="Inbound to Ferry Plaza">
      <prediction epochTime="1588394902295" seconds="11" minutes="0" isDeparture="false" dirTag="14___I_F00" vehicle="6591" block="1631" tripTag="9359290" />
      <prediction epochTime="1588395526282" seconds="635" minutes="10" isDeparture="false" dirTag="14___I_F00" vehicle="6539" block="1633" tripTag="9359291" />
    </direction>
    <message text="Service frequency for this stop is 10 minutes or less." priority="Normal"/>
    <message text="Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info." priority="Low"/>
    <message text="Text COVID19SF to 888-777 for official updates. Visit sfdph.org" priority="Low"/>
    <message text="Board thru rear doors. Front door for accessibility needs only." priority="Normal"/>
  </predictions>
</body>
`;
