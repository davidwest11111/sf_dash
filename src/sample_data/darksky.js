export const SAMPLE_HOURLY_FORECAST = {
  time: 1563930000,
  summary: 'Partly Cloudy',
  icon: 'partly-cloudy-day',
  precipIntensity: 0,
  precipProbability: 0.4, // this was hacked in for testing purposes
  temperature: 61.83,
  apparentTemperature: 61.83,
  dewPoint: 55.15,
  humidity: 0.79,
  pressure: 1014.92,
  windSpeed: 11.74,
  windGust: 17.1,
  windBearing: 275,
  cloudCover: 0.15,
  uvIndex: 2,
  visibility: 10,
  ozone: 303.4,
};

export const SAMPLE_DAILY_FORECAST = {
  time: 1563692400,
  summary: 'Partly cloudy throughout the day.',
  icon: 'partly-cloudy-day',
  sunriseTime: 1563714332,
  sunsetTime: 1563766133,
  moonPhase: 0.66,
  precipIntensity: 0.0001,
  precipIntensityMax: 0.0017,
  precipIntensityMaxTime: 1563771600,
  precipProbability: 0.09,
  precipType: 'rain',
  temperatureHigh: 65.95,
  temperatureHighTime: 1563757200,
  temperatureLow: 56.69,
  temperatureLowTime: 1563800400,
  apparentTemperatureHigh: 65.95,
  apparentTemperatureHighTime: 1563757200,
  apparentTemperatureLow: 56.69,
  apparentTemperatureLowTime: 1563800400,
  dewPoint: 54.13,
  humidity: 0.81,
  pressure: 1013.6,
  windSpeed: 6.05,
  windGust: 12.09,
  windGustTime: 1563742800,
  windBearing: 258,
  cloudCover: 0.47,
  uvIndex: 10,
  uvIndexTime: 1563739200,
  visibility: 8.265,
  ozone: 305.1,
  temperatureMin: 55.82,
  temperatureMinTime: 1563721200,
  temperatureMax: 65.95,
  temperatureMaxTime: 1563757200,
  apparentTemperatureMin: 55.82,
  apparentTemperatureMinTime: 1563721200,
  apparentTemperatureMax: 65.95,
  apparentTemperatureMaxTime: 1563757200,
};

export const SAMPLE_ALERT = {
  title: 'Flood Watch for Mason, WA',
  time: 1509993360,
  expires: 1510036680,
  description: '...FLOOD WATCH REMAINS IN EFFECT THROUGH LATE MONDAY NIGHT...\n',
  uri: 'http://alerts.weather.gov/cap/wwacapget.php?x=WA1255E4DB8494.FloodWatch.1255',
};
