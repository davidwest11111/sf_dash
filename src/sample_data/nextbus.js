// The data returned is actually in XML, this is just what the X2JS library outputs

export const SAMPLE_PREDICTION = {
  direction: {
    prediction: [
      {
        _epochTime: '1588198229803',
        _seconds: '296',
        _minutes: '4',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6609',
        _block: '1620',
        _tripTag: '9359239',
        $$hashKey: 'object:19',
      },
      {
        _epochTime: '1588198681179',
        _seconds: '747',
        _minutes: '12',
        _isDeparture: 'false',
        _dirTag: '14___I_F00',
        _vehicle: '6599',
        _block: '1622',
        _tripTag: '9359240',
        $$hashKey: 'object:20',
      }, { _epochTime: '1588199412858', _seconds: '1479', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6547', _block: '1633', _tripTag: '9359237', $$hashKey: 'object:21' }, { _epochTime: '1588199419316', _seconds: '1486', _minutes: '24', _isDeparture: 'false', _dirTag: '14___I_F00', _vehicle: '6533', _block: '1628', _tripTag: '9359243', $$hashKey: 'object:22' }],
    _title: 'Inbound to Ferry Plaza',
  },
  message: [
    {
      _text: 'Service frequency for this stop is 10 minutes or less.',
      _priority: 'Normal',
    },
    {
      _text: 'Muni for essential trips only. Masks or face coverings are required. Practice physical distancing. Visit sfmta.com/COVID19 for more info.',
      _priority: 'Low',
    }, { _text: 'Text COVID19SF to 888-777 for official updates. Visit sfdph.org', _priority: 'Low' }, { _text: 'Board thru rear doors. Front door for accessibility needs only.', _priority: 'Normal' }],
  _agencyTitle: 'SF Muni',
  _routeTitle: '14-Mission',
  _routeTag: '14',
  _stopTitle: 'Mission St & 24th St',
  _stopTag: '5565',
  $$hashKey: 'object:17',
  className: '',
};

// FIXME
export const SAMPLE_FAILED_PREDICTIONS = {

};
