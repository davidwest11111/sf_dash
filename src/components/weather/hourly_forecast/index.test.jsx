import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import { SAMPLE_HOURLY_FORECAST } from '../../../sample_data/darksky';
import HourlyForecast from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let hourlyForecasts;

  beforeEach(() => {
    hourlyForecasts = [
      { ...SAMPLE_HOURLY_FORECAST },
      { ...SAMPLE_HOURLY_FORECAST },
      { ...SAMPLE_HOURLY_FORECAST },
    ];
  });

  test('(default)', () => {
    wrapper = shallow((
      <HourlyForecast hourlyForecasts={[]} />
    ));
  });

  test('no hourlyForecasts', () => {
    wrapper = shallow((
      <HourlyForecast hourlyForecasts={hourlyForecasts} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
