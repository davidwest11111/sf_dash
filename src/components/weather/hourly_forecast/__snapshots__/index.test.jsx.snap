// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<Table
  className="mb-0"
  responsiveTag="div"
  tag="table"
>
  <thead>
    <tr>
      <th>
        Time
      </th>
      <th>
        Condition
      </th>
      <th
        data-toggle="tooltip"
        title="Temp/Feels Like"
      >
        Temp/FL
      </th>
      <th>
        UVI
      </th>
      <th>
        POP
      </th>
      <th>
        Humid
      </th>
    </tr>
  </thead>
  <tbody />
</Table>
`;

exports[`outputs the expected tree when no hourlyForecasts 1`] = `
<Table
  className="mb-0"
  responsiveTag="div"
  tag="table"
>
  <thead>
    <tr>
      <th>
        Time
      </th>
      <th>
        Condition
      </th>
      <th
        data-toggle="tooltip"
        title="Temp/Feels Like"
      >
        Temp/FL
      </th>
      <th>
        UVI
      </th>
      <th>
        POP
      </th>
      <th>
        Humid
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="hourly-forecast-1563930000"
    >
      <td>
        6PM
      </td>
      <td>
        <WeatherIcon
          className=""
          icon="partly-cloudy-day"
        />
         
        Partly Cloudy
      </td>
      <td>
        62
        /
        62
        °
      </td>
      <td>
        2
      </td>
      <td
        className="text-info"
      >
        40
        %
      </td>
      <td>
        79
        %
      </td>
    </tr>
    <tr
      key="hourly-forecast-1563930000"
    >
      <td>
        6PM
      </td>
      <td>
        <WeatherIcon
          className=""
          icon="partly-cloudy-day"
        />
         
        Partly Cloudy
      </td>
      <td>
        62
        /
        62
        °
      </td>
      <td>
        2
      </td>
      <td
        className="text-info"
      >
        40
        %
      </td>
      <td>
        79
        %
      </td>
    </tr>
    <tr
      key="hourly-forecast-1563930000"
    >
      <td>
        6PM
      </td>
      <td>
        <WeatherIcon
          className=""
          icon="partly-cloudy-day"
        />
         
        Partly Cloudy
      </td>
      <td>
        62
        /
        62
        °
      </td>
      <td>
        2
      </td>
      <td
        className="text-info"
      >
        40
        %
      </td>
      <td>
        79
        %
      </td>
    </tr>
  </tbody>
</Table>
`;
